File: Be-siukat-new/siukat-service-dev_snmptn_2022/routes/fakultas.js
var express = require('express');
var router = express.Router();
var Fakultas = require('../models/fakultas');
var db = require('../config/database');

router.get('/all', function(req, res, next){
  return db.transaction(function (t) {
    Fakultas.findAll().then(fakultas => {
      res.send(fakultas)
    })
  }).catch(function (err) {
    res.status(500).json(err)
  })
})

router.get('/id/:id', function(req, res, next){
  return db.transaction(function (t) {
    Fakultas.findById(req.params.id).then(fakultas => {
      if(fakultas){
        res.send(fakultas)
      }else{
        res.send({"msg": "data tidak ditemukan"})
      }
    })
  }).catch(function (err) {
    res.status(500).json(err)
  })
})

module.exports = router;

File: BackEnd-Siukat/routes/fakultas.go
package routes

import (
	"BackEnd-Siukat/config"
	"BackEnd-Siukat/models"
	"net/http"

	"github.com/gin-gonic/gin"
)

func FakultasRoutes(r *gin.RouterGroup) {
	group := r.Group("/fakultas")
	group.GET("/all", func(c *gin.Context) {
		var data []models.Fakultas
		if err := config.DB.Find(&data).Error; err != nil {
			c.JSON(http.StatusInternalServerError, err)
			return
		}
		c.JSON(http.StatusOK, data)
	})

	group.GET("/id/:id", func(c *gin.Context) {
		id := c.Param("id")
		var data models.Fakultas
		if err := config.DB.Where("kode = ?", id).First(&data).Error; err != nil {
			c.JSON(http.StatusOK, gin.H{"msg": "data tidak ditemukan"})
			return
		}
		c.JSON(http.StatusOK, data)
	})
}
